src/libmongoc/src/mongoc/mongoc-async-cmd.c

```diff
@@ -134,20 +135,40 @@ mongoc_async_cmd_run (mongoc_async_cmd_t *acmd)
 }
 
 void
-_mongoc_async_cmd_init_send (mongoc_async_cmd_t *acmd, const char *dbname)
+_mongoc_async_cmd_init_send (const mongoc_opcode_t cmd_opcode,
+                             mongoc_async_cmd_t *acmd,
+                             const char *dbname)
 {
-   acmd->ns = bson_strdup_printf ("%s.$cmd", dbname);
-
    acmd->rpc.header.msg_len = 0;
    acmd->rpc.header.request_id = ++acmd->async->request_id;
    acmd->rpc.header.response_to = 0;
-   acmd->rpc.header.opcode = MONGOC_OPCODE_QUERY;
-   acmd->rpc.query.flags = MONGOC_QUERY_SECONDARY_OK;
-   acmd->rpc.query.collection = acmd->ns;
-   acmd->rpc.query.skip = 0;
-   acmd->rpc.query.n_return = -1;
-   acmd->rpc.query.query = bson_get_data (&acmd->cmd);
-   acmd->rpc.query.fields = NULL;
+
+// JFW: DELETEME: this appears to be needed by both OPCODE_QUERY and OPCODE_MSG:
+// JFW: Evergreen shows this leaking, but we do destroy it in mongoc_async_cmd_destroy(). Never reached?
```

Comment:
I think this is a red herring and is ok. (At least, I don't see it with memcheck.)

Reply:
Do you have a link to the failing task? The task failures linked to the PR [ASAN variant](https://spruce.mongodb.com/version/61f47cd43e8e86482507bd11/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&variant=asan-ubuntu) do not indicate leaks and look like occurrences of CDRIVER-4268.